src/main/java/org/broadinstitute/hellbender/utils/help/GATKWDLWorkUnitHandler.java

```diff
@@ -135,58 +143,142 @@ protected void processPositionalArguments(
-            final String argType = (String) positionalArgs.get("type");
-            positionalArgs.put("testValue",
-                    getInputValueForTest(
-                            argType,
-                            (String) positionalArgs.get("defaultValue"))
+            final Map<String, Object> positionalArgs = args.get(TemplateProperties.ARGUMENTS_POSITIONAL).get(0);
+            positionalArgs.put(WDL_TEST_VALUE_PROPERTY,
+                    testValueAsJSON(
+                            WDLWorkUnitHandler.POSITIONAL_ARGS,
+                            clp.getPositionalArgumentDefinition(),
+                            (String) positionalArgs.get(TemplateProperties.ARGUMENT_TYPE),
+                            (String) positionalArgs.get(TemplateProperties.ARGUMENT_DEFAULT_VALUE))
             );
         }
     }
 
     /**
-     * Return a test input value for use in the WDL validation test inputs. If an option has WDL type
-     * File, then we need to provide the name of an actual file that exists so cromwell can localize it.
-     * "dummyWDLTestFileName" is a file that will be created by the test task.
+     * Return a test input value for use in the WDL validation test inputs.
+     *
+     * If an option has WDL type File, then we need to provide the name of an actual file that exists so cromwell
+     * can localize it: "dummyWDLTestFileName" is a file that will be created by the test task.
+     *
+     * @param longName the long name for this arg
+     * @param argDef the ArgumentDefinition for this arg
      * @param wdlType the wdl type for which an input value is needed
-     * @param defaultValue the default value for the argument for which an input value is required
+     * @param defaultWDLValue the default value for the argument for which an input value is required
      * @return a test input value that is either the default value, or the name of an actual test file
      * that will exist at test execution time
      */
-    protected String getInputValueForTest(final String wdlType, final String defaultValue) {
+    protected String testValueAsJSON(
+            final String longName,
+            final ArgumentDefinition argDef,
+            final String wdlType,
+            final String defaultWDLValue) {
+        final Argument argumentAnnotation = argDef.getUnderlyingField().getAnnotation(Argument.class);
+        final PositionalArguments positionalAnnotation = argDef.getUnderlyingField().getAnnotation(PositionalArguments.class);
+        final boolean isRequired =
+                (argumentAnnotation != null && !argumentAnnotation.optional())
+                || positionalAnnotation != null;
+        final String dummyWDLTestFile = ((GATKWDLDoclet) getDoclet()).getBuildDir() + "/" + dummyWDLTestFileName;
+
+        // Hack to resolve the mutex argument in GATKSparkTool; otherwise all spark tools will fail since all
+        // mutex args will have a value
```

Comment:
Can you explain more precisely what the problem is here, and how we can avoid triggering it in other tools?

Reply:
This is strictly to accommodate a limitation in the test-value generation code, not something fundamental about the actual WDL or the tool. The test generation code isn't mutex-arg-aware, so it will generate and populate test values for all of the args in a mutex group, which will fail when run - this is a special case to accommodate the one mutex arg that is in all Spark tools to allow those WDLs to run. There should be a more general-purpose solution though. I'll make a ticket to resolve that separately.